evm/src/executor/inspector/cheatcodes/env.rs

```diff
@@ -161,8 +161,11 @@ fn prank(
 ) -> Result {
     let prank = Prank::new(prank_caller, prank_origin, new_caller, new_origin, depth, single_call);
 
-    if let Some(Prank { used, .. }) = state.prank {
+    if let Some(Prank { used, single_call, .. }) = state.prank {
         ensure!(used, "You cannot overwrite `prank` until it is applied at least once");
+        // This case can only fail if the user calls `vm.startPrank` and then `vm.prank` later on.
+        // This should not be possible.
+        ensure!(single_call == single_call, "You cannot override an ongoing prank with a single vm.prank. Use vm.startPrank to override the current prank.");
```

Comment:
is this correct? `single_call == single_call` seems like no-op that will always evaluate to true?

Reply:
yep seems like a no-op haha, but actually, this will evaluate to false if you use `vm.startPrank`, and then use `vm.prank` (which is what surfaced this bug/unintended feature)

Reply:
oh right, I misread that, this indeed looks like a noop

Reply:
yep it's a no-op actually—it's being shadowed by the other `single_call`

Reply:
fixed in https://github.com/foundry-rs/foundry/pull/5520/commits/1c5375e85e898c6ba542bb5868b846a46b147e41